solarforecastarbiter/io/fetch/nwp.py

```diff
@@ -0,0 +1,669 @@
+"""
+Fetch NWP files from NCEP Nomads for select variables. Should primarily be used
+as a CLI program.
+"""
+import asyncio
+import argparse
+from functools import partial
+import logging
+from pathlib import Path
+import re
+import shutil
+import signal
+import subprocess
+import sys
+import tempfile
+
+
+import aiohttp
+import pandas as pd
+import xarray as xr
+
+
+from solarforecastarbiter.io.fetch import (
+    handle_exception, basic_logging_config, make_session,
+    run_in_executor, start_cluster, abort_all_on_exception)
+
+
+logger = logging.getLogger(__name__)
+
+
+CHECK_URL = 'https://nomads.ncep.noaa.gov/pub/data/nccf/com/{}/prod'
+BASE_URL = 'https://nomads.ncep.noaa.gov/cgi-bin/'
+DOMAIN = {'subregion': '',
+          'leftlon': -126,
+          'rightlon': -66,
+          'toplat': 50,
+          'bottomlat': 24}
+
+
+def _gfs_valid_hr_gen(init_hr):
+    i = 0  # should we start at 1 and avoid the analysis?
+    while True:
+        yield i
+        if i < 120:
+            i += 1
+        elif i >= 120 and i < 240:
+            i += 3
+        elif i >= 240 and i < 384:
+            i += 12
+        else:
+            break
+
+
+GFS_0P25_1HR = {'endpoint': 'filter_gfs_0p25_1hr.pl',
+                'file': 'gfs.t{init_hr:02d}z.pgrb2.0p25.f{valid_hr:03d}',
+                'dir': '/gfs.{init_dt}',
+                'lev_2_m_above_ground': 'on',
+                'lev_10_m_above_ground': 'on',
+                'lev_entire_atmosphere': 'on',
+                'lev_surface': 'on',
+                'var_DSWRF': 'on',
+                'var_TCDC': 'on',
+                'var_TMP': 'on',
+                'var_UGRD': 'on',
+                'var_VGRD': 'on',
+                'update_freq': '6h',
+                'valid_hr_gen': _gfs_valid_hr_gen,
+                'time_between_fcst_hrs': 60,
+                'delay_to_first_forecast': '200min',
```

Comment:
This relates to file availability, right? I suggest a brief doc string at the top that lays out the patterns in the file and mentions sources of confusion like this.